geode-core/src/test/java/org/apache/geode/cache/query/security/RegExMethodAuthorizerTest.java

```diff
@@ -0,0 +1,289 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.EntryEvent;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.configuration.RegionConfig;
+import org.apache.geode.internal.cache.BucketDump;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.internal.cache.LocalRegion;
+import org.apache.geode.internal.cache.PartitionedRegion;
+import org.apache.geode.internal.cache.ha.HAContainerMap;
+import org.apache.geode.internal.cache.persistence.query.mock.Pair;
+import org.apache.geode.internal.security.SecurityService;
+import org.apache.geode.internal.util.Valuable;
+import org.apache.geode.security.NotAuthorizedException;
+import org.apache.geode.security.ResourcePermission;
+import org.apache.geode.test.junit.categories.SecurityTest;
+
+@Category(SecurityTest.class)
+public class RegExMethodAuthorizerTest {
+  private InternalCache mockCache;
+  private SecurityService mockSecurityService;
+  private RegExMethodAuthorizer geodeRegExMatchingMethodAuthorizer;
+
+  @Before
+  public void setUp() {
+    mockCache = mock(InternalCache.class);
+    mockSecurityService = mock(SecurityService.class);
+    when(mockCache.getSecurityService()).thenReturn(mockSecurityService);
+    geodeRegExMatchingMethodAuthorizer =
+        new RegExMethodAuthorizer(new RestrictedMethodAuthorizer(mockCache),
+            Collections.singleton("^org\\.apache\\.geode\\..*"));
+  }
+
+  @Test
+  public void constructorShouldThrowExceptionWhenCacheIsNull() {
+    assertThatThrownBy(() -> new RegExMethodAuthorizer((Cache) null, Collections.emptySet()))
+        .isInstanceOf(NullPointerException.class)
+        .hasMessage(NULL_CACHE_MESSAGE);
+  }
+
+  @Test
+  public void constructorShouldThrowExceptionWhenRestrictedMethodAuthorizerIsNull() {
+    assertThatThrownBy(
+        () -> new RegExMethodAuthorizer((RestrictedMethodAuthorizer) null, Collections.emptySet()))
+            .isInstanceOf(NullPointerException.class)
+            .hasMessage(NULL_AUTHORIZER_MESSAGE);
+  }
+
+  @Test
+  public void constructorShouldThrowExceptionWhenAllowedPatternsIsNull() {
+    assertThatThrownBy(() -> new RegExMethodAuthorizer(mockCache, null))
+        .isInstanceOf(NullPointerException.class)
+        .hasMessage(NULL_REGULAR_EXPRESSIONS_MESSAGE);
+
+    assertThatThrownBy(
+        () -> new RegExMethodAuthorizer(new RestrictedMethodAuthorizer(mockCache), null))
+            .isInstanceOf(NullPointerException.class)
+            .hasMessage(NULL_REGULAR_EXPRESSIONS_MESSAGE);
+  }
+
+  @Test
+  public void allowedPatternsShouldBeAccessibleBuNotModifiable() {
+    assertThatThrownBy(() -> geodeRegExMatchingMethodAuthorizer.getAllowedPatterns().remove(".*"))
```

Comment:
The pattern attempting to be removed doesn't appear to match the pattern that was used to initialize the Authorizer on line 76. Perhaps it should be extracted to a field if it's going to be referenced in this method too.

Reply:
The method is just verifying that the original set can no be modified (`remove` throws an `UnsupportedOperationException`, no matter whether the collection contains the element or not). Trying to remove something that doesn't even exist might confuse the reader as you say, though, so I'll change the method to try to to add something to the set instead of removing it (the result will be the same anyway -> the collection can't be modified).